src/device/dcd.h

```diff
@@ -154,22 +135,6 @@ void dcd_edpt_stall       (uint8_t rhport, uint8_t ep_addr);
 void dcd_edpt_clear_stall (uint8_t rhport, uint8_t ep_addr);
 bool dcd_edpt_stalled     (uint8_t rhport, uint8_t ep_addr);
 
-//------------- Control Endpoint -------------//
-bool dcd_control_xfer     (uint8_t rhport, uint8_t dir, uint8_t * buffer, uint16_t length);

```

Comment:
why would you want to merge the control xfer into the normal transfer. Although it is similar, control pipe is bidirectional, the direction is needed without requiring dcd to guess/cached which direction stack trying to carry

Reply:
the endpoint 0 is bidirectional, other endpoin can only be either IN or OUT. But indeed merging them is good way. It will require usbd to pass 0x80 as control IN address, 0x00 for control OUT. Note: 0x80 for control IN is not part of usb specs, but we could document for dcd to take that.